Source/Libraries/openXDA.Model/Channels/ChannelData.cs

```diff
@@ -289,6 +309,67 @@ private static byte[] ToData(List<DataPoint> data, int seriesID)
             return returnArray;
         }
 
+        private static byte[] ToDigitalData(List<DataPoint> data, int seriesID)
+        {
+            int i;
+            IEnumerable<DataPoint> changeSeries = data.Where((point,index) => index == 0 || index == data.Count - 1 || point.Value != data[index-1].Value);
+
+            const ushort NaNValue = ushort.MaxValue;
+            const ushort MaxCompressedValue = ushort.MaxValue - 1;
+            double range = changeSeries.Select(item => item.Value).Max() - changeSeries.Select(item => item.Value).Min();
+            double decompressionOffset = changeSeries.Select(item => item.Value).Min();
+            double decompressionScale = range / MaxCompressedValue;
+            double compressionScale = (decompressionScale != 0.0D) ? 1.0D / decompressionScale : 0.0D;
```

Comment:
I think we could simplify the logic quite a bit if we assume digitals can only be 0 or 1. We could compress the whole digital series down to just the first and last data values, the first and last timestamps, and the timestamp of each state change.

Reply:
Maybe, but I think we should avoid restraining this to `0` and `1` only
I left it open ended and in theory a latched analog may also be compressed this way since it would not change value more than 10% of the time.
Ultimately the reason for this is not to compress the data but to speed up visualization - it's just that we do that by compressing data here. I don't think that needs to be optimized too much